include/swift/AST/Expr.h

```diff
@@ -4867,6 +4871,13 @@ inline bool Expr::isInfixOperator() const {
          isa<AssignExpr>(this) || isa<ExplicitCastExpr>(this);
 }
 
+inline bool Expr::needsParensInsidePostfixOperator() const {
+  // Prefix and infix operators will bind outside of a postfix operator.
+  // Postfix operators will get token-merged with a new postfix operator.
+  return (isa<PrefixUnaryExpr>(this) || isa<PostfixUnaryExpr>(this) ||
+          isa<OptionalEvaluationExpr>(this) || this->isInfixOperator());
```

Comment:
…which is important because this is definitely missing cases (casts, for example).